### PR TITLE
CA-208836

### PR DESCRIPTION
**Pull request by jonludlam**
Don't try to enable a licensed feature by default.

### CODE REVIEW

File: ocaml/idl/datamodel.ml [hunk 1]
```diff
@@ -7380,7 +7380,13 @@ let vm =
 	field ~qualifier:StaticRO ~in_product_since:rel_boston ~default_value:(Some (VInt 0L)) ~ty:Int "version" "The number of times this VM has been recovered";
 	field ~qualifier:StaticRO ~in_product_since:rel_clearwater ~default_value:(Some (VString "0:0")) ~ty:(String) "generation_id" "Generation ID of the VM";
 	field ~writer_roles:_R_VM_ADMIN ~qualifier:RW ~in_product_since:rel_cream ~default_value:(Some (VInt 0L)) ~ty:Int "hardware_platform_version" "The host virtual hardware platform version the VM can run on";
-	field ~qualifier:StaticRO ~lifecycle:[Published, rel_dundee, ""] ~doc_tags:[Windows] ~default_value:(Some (VBool true)) ~ty:Bool "has_vendor_device" "When an HVM guest starts, this controls the presence of the emulated C000 PCI device which triggers Windows Update to fetch or update PV drivers.";
+ field ~qualifier:StaticRO ~lifecycle:[Published, rel_dundee, ""] ~doc_tags:[Windows] ~default_value:(Some (VCustom (String.concat "\n" [
+     "(Rpc.Bool (";
+     "let pool = List.hd (Db_actions.DB_Action.Pool.get_all ~__context) in";
```
Comment (robhoes) [inline ocaml/idl/datamodel.ml]: `try .. with`?

File: ocaml/idl/datamodel.ml [hunk 2]
```diff
@@ -7379,7 +7380,13 @@ let vm =
...
 	field ~qualifier:StaticRO ~in_product_since:rel_clearwater ~default_value:(Some (VString "0:0")) ~ty:(String) "generation_id" "Generation ID of the VM";
 	field ~writer_roles:_R_VM_ADMIN ~qualifier:RW ~in_product_since:rel_cream ~default_value:(Some (VInt 0L)) ~ty:Int "hardware_platform_version" "The host virtual hardware platform version the VM can run on";
-	field ~qualifier:StaticRO ~lifecycle:[Published, rel_dundee, ""] ~doc_tags:[Windows] ~default_value:(Some (VBool true)) ~ty:Bool "has_vendor_device" "When an HVM guest starts, this controls the presence of the emulated C000 PCI device which triggers Windows Update to fetch or update PV drivers.";
+ field ~qualifier:StaticRO ~lifecycle:[Published, rel_dundee, ""] ~doc_tags:[Windows] ~default_value:(Some (VCustom (String.concat "\n" [
+     "(try Rpc.Bool (";
+     "let pool = List.hd (Db_actions.DB_Action.Pool.get_all ~__context) in";
+     "let restrictions = Db_actions.DB_Action.Pool.get_restrictions ~__context ~self:pool in ";
+     "let vendor_device_allowed = try List.assoc \"restrict_pci_device_for_auto_update\" restrictions = \"false\" with _ -> false in";
```
Comment (robhoes) [inline ocaml/idl/datamodel.ml]: Actually, this would be better: `Pool_features.is_enabled ~__context Features.PCI_device_for_auto_update`.

Comment (jonludlam) [inline ocaml/idl/datamodel.ml]: Ah yes, that would be better.

Comment (jonludlam) [inline ocaml/idl/datamodel.ml]: Ah actually no, I don't have access to the xapi modules, they're not in scope when this is compiled. I think it'll have to be left as it is.

**Pull request merged**